tools/firmware-utils/src/tplink-safeloader.c

```diff
@@ -2207,6 +2207,47 @@ static struct device_info boards[] = {
 		.last_sysupgrade_partition = "file-system"
 	},
 
+	/** Firmware layout for the RE305 v3 */
+	{
+		.id     = "RE305-V3",
+		.vendor = "",
+		.support_list =
+			"SupportList:\n"
+			"{product_name:RE305,product_ver:3.0.0,special_id:00000000}\n"
+			"{product_name:RE305,product_ver:3.0.0,special_id:45550000}\n"
+			"{product_name:RE305,product_ver:3.0.0,special_id:4A500000}\n"
+			"{product_name:RE305,product_ver:3.0.0,special_id:4B520000}\n"
+			"{product_name:RE305,product_ver:3.0.0,special_id:41550000}\n"
+			"{product_name:RE305,product_ver:3.0.0,special_id:42520000}\n"
+			"{product_name:RE305,product_ver:3.0.0,special_id:55530000}\n"
+			"{product_name:RE305,product_ver:3.0.0,special_id:45530000}\n"
+			"{product_name:RE305,product_ver:3.0.0,special_id:41530000}\n"
+			"{product_name:RE305,product_ver:3.0.0,special_id:43410000}\n"
+			"{product_name:RE305,product_ver:3.0.0,special_id:52550000}\n",
+		.part_trail = 0x00,
+		.soft_ver = "soft_ver:2.0.0\n",
```

Comment:
This needs adapting after 8dba4741b3 - should become:

`.soft_ver = SOFT_VER_TEXT("soft_ver:2.0.0\n"),`